src/vm/memory/segments.zig

```diff
@@ -168,7 +168,25 @@ pub const MemorySegmentManager = struct {
     /// # Returns
     ///
     /// An `AutoArrayHashMap` representing the computed effective sizes of memory segments.
-    pub fn computeEffectiveSize(self: *Self) !std.AutoArrayHashMap(u32, u32) {
+    pub fn computeEffectiveSize(self: *Self, allow_tmp_segments: bool) !std.AutoArrayHashMap(i64, u32) {
```

Comment:
Can we also add unit tests for `allow_tmp_segments = true` in order to validate this new feature.

Reply:
> try to check is the obtained size the all the segments are correct with the allow_tmp_segments flag set to true.

that is what i dont know, is there any good doc to understand how memory segments work?

Reply:
You can look in the unit tests that are already done, for example this one: "MemorySegmentManager: computeEffectiveSize for one segment memory with gaps".

When you have:
```zig
try memoryFile.setUpMemory(
         memory_segment_manager.memory,
         std.testing.allocator,
         .{
             .{ .{ 0, 3 }, .{1} },
             .{ .{ 0, 4 }, .{1} },
             .{ .{ 0, 7 }, .{1} },
             .{ .{ 0, 9 }, .{1} },
         },
     );
```

For now you have:
```zig
var actual = try memory_segment_manager.computeEffectiveSize();
try expectEqual(@as(usize, 1), actual.count());
try expectEqual(@as(u32, 10), actual.get(0).?);
```

You can test the same thing by adding negative indexes by testing activating the `allow_tmp_segments ` flag.